modules/tinymce/src/core/main/ts/selection/MultiClickSelection.ts

```diff
@@ -1,44 +1,83 @@
-import { SugarElement } from '@ephox/sugar';
+import { Compare, PredicateFind, SugarElement } from '@ephox/sugar';
 
+import DomTreeWalker from '../api/dom/TreeWalker';
 import Editor from '../api/Editor';
-import * as CaretFinder from '../caret/CaretFinder';
+import { isCaretCandidate } from '../caret/CaretCandidate';
 import { CaretPosition } from '../caret/CaretPosition';
-import { isTextBlock } from '../dom/ElementType';
+import { isBlock } from '../dom/ElementType';
 import * as NodeType from '../dom/NodeType';
 import * as RangeNormalizer from './RangeNormalizer';
 
-const isTextBlockNode = (node: Node): node is Element => NodeType.isElement(node) && isTextBlock(SugarElement.fromDom(node));
+const isBr = NodeType.isBr;
+const isText = NodeType.isText;
+const isContentEditableFalse = (elm: SugarElement<Node>): boolean => NodeType.isContentEditableFalse(elm.dom);
+const isContentEditableTrue = (elm: SugarElement<Node>): boolean => NodeType.isContentEditableTrue(elm.dom);
+const isRoot = (rootNode: Node) => (elm: SugarElement<Node>): boolean => Compare.eq(SugarElement.fromDom(rootNode), elm);
```

Comment:
I wonder how many times we have invented this function maybe we should add some tech dept ticket to just include this partial function in sugar Compare. This is more of a general comment. I think Hannes also invented this function in the other PR. :smile:

Reply:
It's probably a lot... the problem with this one is that we need that shared library for TinyMCE and other monorepo libs as we have nowhere to place this sort of thing atm and we can't create a circular reference.